pkg/apis/apis/v1alpha1/types.go

```diff
@@ -0,0 +1,375 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/third_party/conditions/apis/conditions/v1alpha1"
+)
+
+// APIBinding enables a set of resources and their behaviour through an external
+// service provider in this workspace.
+//
+// The service provider uses an APIExport to expose the API.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+type APIBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	// +optional
+	Spec APIBindingSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state.
+	// +optional
+	Status APIBindingStatus `json:"status,omitempty"`
+}
+
+// APIBindingSpec records the APIs and implementations that are to be bound.
+type APIBindingSpec struct {
+	// reference uniquely identifies an API to bind to.
+	//
+	// +optional
+	Reference ExportReference `json:"reference,omitempty"`
+}
+
+// ExportReference describes a reference to an APIExport. Exactly one of the
+// fields must be set.
+type ExportReference struct {
+	// workspace is a reference to an APIExport in the same organization. The creator
+	// of the APIBinding needs to have access to the APIExport with the verb `bind`
+	// in order to bind to it.
+	//
+	// +optional
+	Workspace *WorkspaceExportReference `json:"workspace,omitempty"`
+}
+
+// WorkspaceExportReference describes an API and backing implementation that are provided by an actor in the
+// same Workspace as the ServiceBinding.
+type WorkspaceExportReference struct {
+	// name is a workspace name in the same organization.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kube:validation:MinLength=1
+	WorkspaceName string `json:"name"`
+
+	// Name of the APIExport that describes the API.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kube:validation:MinLength=1
+	ExportName string `json:"exportName"`
+}
+
+// APIBindingPhaseType is the type of the current phase of an APIBinding.
+type APIBindingPhaseType string
+
+const (
+	APIBindingPhaseInitializing APIBindingPhaseType = "Initializing"
```

Comment:
This is a mismatch with the phase enum below (Binding, not Initializing). Needs to match (either way)

Reply:
Binding it is